### ISSUE TITLE
Cannot execute final step (archive)

### ISSUE THREAD

**Issue by dayaketu** [pieye/toggl2clockify#13]
Hello!

I'm on windows 10, and cannot finalize the last step of the migration mentioned on the guide, which is referring to archive. Here are the logs. Can you please suggest next steps?

2019-11-10 20:50:00,491 - INFO - toggl2clockify - 'EndTime' not found in config file importing all entries until now

2019-11-10 20:50:00,494 - INFO - Clue - testing toggl API key XXXXXXXXXX
2019-11-10 20:50:02,935 - INFO - Clue - ...ok, togglKey resolved to email XXXXXXXXXX
2019-11-10 20:50:02,936 - INFO - ClockifyAPI - testing clockify APIKey XXXXXXXXXX
2019-11-10 20:50:03,419 - INFO - ClockifyAPI - ...ok, key resolved to email XXXXXXXXXX
2019-11-10 20:50:03,420 - INFO - ClockifyAPI - testing clockify APIKey XXXXXXXXXX
2019-11-10 20:50:03,701 - INFO - ClockifyAPI - ...ok, key resolved to email XXXXXXXXXX
2019-11-10 20:50:04,371 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,371 - INFO - toggl2clockify - Starting to import workspace 'EpicLemon' (1 of 1)
2019-11-10 20:50:04,373 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,373 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,374 - INFO - toggl2clockify - Phase 1 of 7: Import clients
2019-11-10 20:50:04,374 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,375 - INFO - toggl2clockify - ... skipping phase 1
2019-11-10 20:50:04,375 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,375 - INFO - toggl2clockify - Phase 1 of 7 (Import clients) completed (entries=0, ok=0, skips=0, err=0)
2019-11-10 20:50:04,376 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,376 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,377 - INFO - toggl2clockify - Phase 2 of 7: Import tags
2019-11-10 20:50:04,377 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,378 - INFO - toggl2clockify - ... skipping phase 2
2019-11-10 20:50:04,379 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,381 - INFO - toggl2clockify - Phase 2 of 7 (Import tags) completed (entries=0, ok=0, skips=0, err=0)
2019-11-10 20:50:04,386 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,386 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:04,387 - INFO - toggl2clockify - Phase 3 of 7: Import groups
2019-11-10 20:50:04,387 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:05,574 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:05,575 - INFO - toggl2clockify - Phase 3 of 7 (Import groups) completed (entries=0, ok=0, skips=0, err=0)
2019-11-10 20:50:05,576 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:05,577 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:05,577 - INFO - toggl2clockify - Phase 4 of 7: Import projects
2019-11-10 20:50:05,579 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:05,582 - INFO - toggl2clockify - ... skipping phase 3
2019-11-10 20:50:05,582 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:05,583 - INFO - toggl2clockify - Phase 4 of 7 (Import projects) completed (entries=0, ok=0, skips=0, err=0)
2019-11-10 20:50:05,584 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:05,585 - INFO - toggl2clockify - -------------------------------------------------------------
2019-11-10 20:50:05,588 - INFO - toggl2clockify - Phase 5 of 7: Import tasks
2019-11-10 20:50:05,588 - INFO - toggl2clockify - -------------------------------------------------------------
Traceback (most recent call last):
  File "toggl2clockify.py", line 261, in <module>
    numEntries, numOk, numSkips, numErr = cl.syncTasks(ws)
  File "Clue.py", line 112, in syncTasks
    tProjs = self.toggl.projects
AttributeError: 'TogglAPI' object has no attribute 'projects'
[144456] Failed to execute script toggl2clockify

**Comment by rdlagency** [comment]
@epiclemon  I had the same and reading through the code, seemed to be caused by importing tasks without projects and there's a flag available of --skipTasks.  Adding that too, skips that block and the archiving runs.   

i.e. python toggl2clockify.py --skipClients --skipProjects --skipEntries --skipTags --skipTasks --doArchive

**Comment by dayaketu** [comment]
@rdlagency Thanks, it worked :)

**Comment by dayaketu** [comment]
I wanted to report back that running that did something wrong. It archived projects on Clockify that were not archived on Toggl and it did not archive on Clockify projects that were archived in Toggl. No error happened or at least was mentioned in the logs.

**Comment by rdlagency** [comment]
@epiclemon Sorry to hear that - mine did work OK, but had to manually catch 50 or so projects at the end for some reason which seemed not to archive.  Hope you manage to resolve.

**Comment by Royedc4** [comment]
I've catch a different error when doing the final step `python toggl2clockify.py --skipClients --skipProjects --skipEntries --skipTags --skipTasks --doArchive`

```
2020-04-01 13:20:41,158 - INFO - Clue - project 0 Androids is not active, trying to archive (5 of 127)
2020-04-01 13:20:42,468 - WARNING - ClockifyAPI - Archiving project 0 Androids failed, status code=404, msg=
2020-04-01 13:20:42,469 - INFO - Clue - project 0 Asd Migration is not active, trying to archive (6 of 127)
2020-04-01 13:20:43,577 - WARNING - ClockifyAPI - Archiving project 0 Asd Migration failed, status code=404, msg=

```

It is not archiving any project

**Comment by alex-ong** [comment]
> i.e. python toggl2clockify.py --skipClients --skipProjects --skipEntries --skipTags --skipTasks --doArchive

This has been added in the documentation in #21 

> I've catch a different error when doing the final step python toggl2clockify.py --skipClients --skipProjects --skipEntries --skipTags --skipTasks --doArchive

@Royedc4 
Archiving API changed, requires PUT request. That's why it's failing. I've patched it in #21 . Hopefully this will be merged soon; otherwise try pulling my branch. If you're using the .exe you're out of luck since it hasn't been updated since 2 years ago.

API for deletion (you probably never used it, but anyway) also changed and requires archive before delete.
